connectors/preflight_check.py

```diff
@@ -124,33 +124,44 @@ def _validate_configuration(self):
             )
 
         # Connector client mode
-        configured_connector_id = self.config.get("connector_id", None)
-        configred_service_type = self.config.get("service_type", None)
+        configured_connectors = self.config.get("connectors", []) or []
+        deprecated_connector_id = self.config.get("connector_id", None)
+        deprecated_service_type = self.config.get("service_type", None)
         if (
-            not (configured_connector_id and configred_service_type)
-            and not force_allowed_native
+            not configured_connectors
+            and deprecated_connector_id
+            and deprecated_service_type
         ):
+            configured_connectors.append(
+                {
+                    "connector_id": deprecated_connector_id,
+                    "service_type": deprecated_service_type,
+                }
+            )
```

Comment:
Make sense. I will add a warning log, but I will not mention the target release as I have no idea

Reply:
I think this is the right way - to say that the deprecation is coming, but not to promise a release. @wangch079 can you also make sure that one of these PRs has a `### Release Note` section and the `release-note` label so that this deprecation gets included in the 8.10 release notes?